src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.Simple.cs

```diff
@@ -18,6 +18,11 @@ public partial class CborWriter
         /// <para>The written data is not accepted under the current conformance mode.</para></exception>
         public void WriteSingle(float value)
         {
+            if (float.IsNaN(value))
```

Comment:
I see that `WriteSingle` and `WriteDouble` are covered, but not `WriteHalf`

https://github.com/dotnet/runtime/blob/88f25474334bfb0da10e0b5acb0219a64a04828b/src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.Simple.netcoreapp.cs#L20-L27

I think the appropriate change is to leave WriteSingle and WriteDouble alone, and fix WriteHalf to say "If this is NaN, then write it with positive NaN"

Reply:
> I think the appropriate change is to leave WriteSingle and WriteDouble alone, and fix WriteHalf to say "If this is NaN, then write it with positive NaN"

It wouldn't cover writing a float/double NaN and requiring preserving precision (they should be positive too as per RFC 7049).

You're right about WriteHalf, though.

Reply:
I pushed the positive NaN reassignment to Write(Single|Double)Core because most of the cases are handled by WriteHalf, that way code for Write(Double|Single) is left alone. Besides, it's more of an representation thing so it's probably better to handle it just before writing.